cdap-event-reader-spi/src/main/java/io/cdap/cdap/spi/events/EventResult.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.events;
+
+import java.util.function.Consumer;
+
+/**
+ * Result of reading event.
+ *
+ * @param <T> Event received
+ */
+public interface EventResult<T extends Event> extends AutoCloseable {
+
+    /**
+     * Process messages
+     *
+     * @param consumer
+     */
+    void consumeMessages(Consumer<T> consumer);
```

Comment:
IMO, this needs to be public method.

Reply:
interface methods are all public, hence no need to have `public` keyword.

Reply:
sg. thanks